src/App.vue

```diff
@@ -342,54 +342,85 @@
 	}
 </style>
 <script>
-import Modal from 'nextcloud-vue/dist/Components/Modal'
+import Modal from '@nextcloud/vue/dist/Components/Modal'
 import axios from '@nextcloud/axios'
+import { loadState } from '@nextcloud/initial-state'
 import { generateUrl } from '@nextcloud/router'
+import IntroVideo from './components/IntroVideo'
+import IntroImage from './components/IntroImage'
+
+let hasVideo = false
+try {
+	hasVideo = loadState('firstrunwizard', 'hasVideo')
+} catch (e) {
+	// ignore missing initial state
+}
 
 export default {
-	name: 'FirstRunWizard',
+	name: 'App',
 	components: {
-		Modal
+		Modal,
 	},
 	data() {
 		return {
 			showModal: false,
+			withIntro: false,
+			hasVideo,
 			slides: [],
 			currentSlide: 0,
 			fadeDirection: 'next',
-			isMobile: window.outerWidth < 768
+			isMobile: window.outerWidth < 768,
 		}
 	},
 	computed: {
+		slideList() {
+			if (this.withIntro) {
+				return this.slides
+			}
+			const slides = this.slides
+			return slides.splice(1)
+		},
 		hasNext() {
-			return this.currentSlide < this.slides.length - 1
+			return this.currentSlide < this.slideList.length - 1
 		},
 		hasPrevious() {
 			return this.currentSlide > 0
 		},
 		isLast() {
-			return this.currentSlide === this.slides.length - 1
+			return this.currentSlide === this.slideList.length - 1
 		},
 		isFirst() {
 			return this.currentSlide === 0
-		}
+		},
 	},
-	created() {
-		const img = new Image()
-		img.src = require('../img/intro.png')
-		img.onload = () => {
-			axios.get(generateUrl('/apps/firstrunwizard/wizard')).then((response) => {
-				this.slides = response.data
-			})
+	async created() {
+		if (hasVideo) {
+			this.slides = [ IntroVideo ]
+		} else {
+			this.slides = [ IntroImage ]
+			const img = new Image()
+			img.src = require('../img/intro.png')
+			img.onload = () => {}
 		}
 		window.addEventListener('resize', this.onResize)
 	},
 	beforeDestroy() {
 		window.removeEventListener('resize', this.onResize)
 	},
 	methods: {
-		open() {
+		async loadStaticSlides() {
+			try {
+				const response = await axios.get(generateUrl('/apps/firstrunwizard/wizard'))
+				this.slides.push(...response.data)
+			} catch (e) {
+				console.error('Failed to load slides')
```

Comment:
Show an error if it fails? Or something? :)

Reply:
I'd say we should just ignore that, since it is mainly when the user opens Nextcloud for the first time. If it fails for some reason we just show the user interface and skip the wizard.

Reply:
Also errors shouldn't be an issue here, we just fetch template files and return them in a route.

Reply:
Good to me! You're the captain here ;)